classes/QueryCondition.cls

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copied and slightly modified from https://www.jamessimone.net/blog/joys-of-apex/repository-pattern/
+ */
+public with sharing class QueryCondition implements Comparable {
+    public enum Operator {
+        EQUALS,
+        NOT_EQUALS,
+        LESS_THAN,
+        LESS_THAN_OR_EQUAL,
+        GREATER_THAN,
+        GREATER_THAN_OR_EQUAL
+    }
+
+    private final SObjectField field;
+    private final Operator operator;
+    private final List<Object> values;
+
+    private static Boolean isSet = false;
+
+    public QueryCondition(SObjectField field, Operator operator, Object value) {
+        this(field, operator, new List<Object>{ value });
+    }
+
+    public QueryCondition(SObjectField field, Operator operator, List<Object> values) {
+        this.field = field;
+        this.operator = operator;
+        this.values = values;
+    }
+
+    public override String toString() {
+        String fieldName = this.field.getDescribe().getName();
+        String valueString = this.convertToString(this.values);
+        return fieldName + ' ' + this.getOperator() + ' ' + valueString;
+    }
+
+    private String convertToString(Object value) {
+        if (value == null) {
+            return 'null';
+        } else if (value instanceof Datetime) {
+            return this.formatDatetime(value);
+        } else if (value instanceof List<Object>) {
```

Comment:
Will this support decimal sets?

Reply:
it should but I haven't tested that yet. sets are special case that are handled after these conditionals